src/Deserializers/EntityDeserializer.php

```diff
@@ -59,6 +59,11 @@ public function deserialize( $serialization ) {
 	 */
 	protected abstract function getPartiallyDeserialized( array $serialization );
 
+	/**
+	 * @param array $serialization
+	 *
+	 * @return Entity
```

Comment:
What do you mean with it not being that easy? What problem occurs when using this type hint?